### DISCUSSION TITLE
How does hosting multiplayer work?

### DISCUSSION THREAD

**Discussion by Tmkrth** [diasurgical/DevilutionX#2885] [Q&A] [answered]
Hello! How does the hosting of the game work?

Let's say there are four people in a TCP/IP game, one of them has created that game. Leaving and re-joining works for the other three people. But when the one who created the game leaves, she does not seem to be able to join any of the other player's IPs, albeit the game continues for the other players.

Should this work, is our configuration not yet error free?

Or is it intended, that the host cannot re-join it's own game?

Thank you

**Comment by AJenbo** [comment, marked as answer]
TCP is not a P2P protocol, the host will be running an echo server that every one connects to, once the host leaves the server is shutdown and everyone is isolated.

0.1.0 and 1.3.0 has option for other network types that do not have this limitation.
